### PR TITLE
Search

### PR DESCRIPTION
**Pull request by PawelDzi**
I did a search engine but it doesn't work properly

### CODE REVIEW

File: src/common/Header/Search/index.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            onChange={inputChange}
+            placeholder={`${location.pathname.includes("movie")
+                ? "Search for movies..."
+                : "Search for people..."
+                }`}
+        />
+    );
+};
+
+export default SearchEngine;
```
Comment (Albrecht-Albi) [inline src/common/Header/Search/index.js:36]: Nice, we have a search engine!!

File: src/features/MoviesList/index.js [hunk 2]
```diff
@@ -1,45 +1,64 @@
...
+import { useSelector, useDispatch } from "react-redux";
+import { selectMoviesList, selectStatus, selectTotalResults, setQuery } from "../moviesListSlice";
 import { MainWrapper } from "../../common/MainWrapper";
 import { MovieTile } from "../../common/Tiles";
 import { List, Item } from "./styled";
 import { Pagination } from "../../common/Pagination";
 import { Error } from "../../common/Error";
 import Loader from "../../common/Loader";
 import { Title } from "../../common/Title";
+import { useQueryParameter } from "../queryParameters";
```
Comment (Albrecht-Albi) [inline src/features/MoviesList/index.js:10]: You used dispatch well

File: src/features/moviesListSlice.js [hunk 3]
```diff
@@ -7,25 +7,34 @@ const moviesListSlice = createSlice({
...
+      state.totalResults = payload.data.total_results;
     },
     fetchMoviesListError: (state) => {
       state.status = "error";
     },
+    setQuery: (state, { payload }) => {
+      state.query = payload.query;
+      state.status = "loading";
+    },
   },
```
Comment (Albrecht-Albi) [inline src/features/moviesListSlice.js:27]: You used query well

File: src/features/moviesListSaga.js [hunk 4]
```diff
@@ -1,15 +1,22 @@
...
     yield delay(1000);
-    const data = yield call(getPopularMovies);
     const genres = yield call(getGenres);
+    const query = yield select(selectQuery);
+    let data;
+    if (query !== '') {
+      data = yield call(searchMovie, { query: query });
+    } else {
+      data = yield call(getPopularMovies);
+    }
```
Comment (lkkolosowski) [inline src/features/moviesListSaga.js:19]: good job Pawel

File: src/features/queryParameters.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+        for (const { key, value } of keyValue) {
+            if (value === undefined) {
+                searchParams.delete(key);
+            } else {
+                searchParams.set(key, value);
+            }
+        }
+        history.push(`${location.pathname.slice(0, 7)}?${searchParams.toString()}`);
+    };
+
```
Comment (Albrecht-Albi) [inline src/features/queryParameters.js:25]: I was missing this file

**Review by Albrecht-Albi — APPROVED** [review]

File: src/features/queryParameters.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    return replaceQueryParameter;
```
Comment (lkkolosowski) [inline src/features/queryParameters.js:26]: You had to work hard to done this 😉

File: src/features/MoviesList/index.js [hunk 7]
```diff
@@ -1,45 +1,64 @@
...
-      />
-      <Pagination />
+      {totalResults === 0 ?
+        <Error /> :
+        (<><MainWrapper
+          content={
+            <>
+              <Title title={
+                query ? `Search results for "${query}" (${totalResults})` :
+                  `Popular Movies`
```
Comment (lkkolosowski) [inline src/features/MoviesList/index.js:39]: This should work great 😊

**Review by lkkolosowski — APPROVED** [review]

**Review by Albrecht-Albi — APPROVED** [review]

**Pull request merged**